diff/diff.go

```diff
@@ -0,0 +1,534 @@
+			for ; index1 < len(rowsData1); index1++ {
+				sql := generateDML("replace", rowsData1[index1], orderKeyCols, table.Info, table.Schema)
+				log.Infof("[insert] sql: %v", sql)
+				df.wg.Add(1)
+				df.sqlCh <- sql
+				equal = false
+			}
+			break
+		}
+		eq, cmp, err := compareData(rowsData1[index1], rowsData2[index2], orderKeyCols)
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+		if eq {
+			index1++
+			index2++
+			continue
+		}
+		equal = false
+		switch cmp {
+		case 1:
+			// delete
+			sql := generateDML("delete", rowsData2[index2], orderKeyCols, table.Info, table.Schema)
+			log.Infof("[delete] sql: %v", sql)
+			df.wg.Add(1)
+			df.sqlCh <- sql
+			index2++
+		case -1:
+			// insert
+			sql := generateDML("replace", rowsData1[index1], orderKeyCols, table.Info, table.Schema)
+			log.Infof("[insert] sql: %v", sql)
+			df.wg.Add(1)
+			df.sqlCh <- sql
+			index1++
+		case 0:
+			// update
+			deleteSQL := generateDML("delete", rowsData2[index2], orderKeyCols, table.Info, table.Schema)
+			replaceSQL := generateDML("replace", rowsData1[index1], orderKeyCols, table.Info, table.Schema)
+			log.Infof("[update] delete: %s,\n replace: %s", deleteSQL, replaceSQL)
+			df.wg.Add(2)
+			df.sqlCh <- deleteSQL
+			df.sqlCh <- replaceSQL
+			index1++
+			index2++
+		}
+	}
+
+	return equal, nil
+}
+
+// WriteSqls write sqls to file
+func (df *Diff) WriteSqls() {
+	defer df.wg.Done()
+	for {
+		select {
+		case dml, ok := <-df.sqlCh:
+			if !ok || dml == "end" {
+				return
+			}
+
+			_, err := df.fixSQLFile.WriteString(fmt.Sprintf("%s\n", dml))
+			if err != nil {
+				log.Errorf("write sql: %s failed, error: %v", dml, err)
+			}
+			df.wg.Done()
+		default:
+		}
+	}
+}
+
+func generateDML(tp string, data map[string][]byte, keys []*model.ColumnInfo, table *model.TableInfo, dbName string) (sql string) {
+	// TODO: can't distinguish NULL between ""
+	switch tp {
+	case "replace":
+		colNames := make([]string, 0, len(table.Columns))
+		values := make([]string, 0, len(table.Columns))
+		for _, col := range table.Columns {
+			colNames = append(colNames, col.Name.O)
+			if needQuotes(col.FieldType) {
+				values = append(values, fmt.Sprintf("\"%s\"", string(data[col.Name.O])))
+			} else {
+				values = append(values, string(data[col.Name.O]))
+			}
+		}
+
+		sql = fmt.Sprintf("REPLACE INTO `%s`.`%s`(%s) VALUES (%s);", dbName, table.Name, strings.Join(colNames, ","), strings.Join(values, ","))
+	case "delete":
+		kvs := make([]string, 0, len(keys))
+		for _, col := range keys {
+			if needQuotes(col.FieldType) {
+				kvs = append(kvs, fmt.Sprintf("%s = \"%s\"", col.Name.O, string(data[col.Name.O])))
+			} else {
+				kvs = append(kvs, fmt.Sprintf("%s = %s", col.Name.O, string(data[col.Name.O])))
+			}
+		}
+		sql = fmt.Sprintf("DELETE FROM `%s`.`%s` where %s;", dbName, table.Name, strings.Join(kvs, " AND "))
+	default:
+		log.Errorf("unknow sql type %s", tp)
+	}
+
+	return
+}
+
+func needQuotes(ft types.FieldType) bool {
```

Comment:
name and function do not match

Reply:
if type is number or float, don't  need quote.